src/PrivateRoute/PrivateRoute.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import { Route, Redirect } from 'react-router-dom'
+
+export default function PrivateRoute({component: Component, ...rest}) {
+  return (
+    <Route
+    {...rest}
+    render={(props) => {
+      if (localStorage.getItem('token') && props.location.pathname !== '/') {
+        return <Component {...props} />
+      }
```

Comment:
better to make it as a HOC. Because you are repeating yourself. 
Also such kind of logic should be handled on the app level. Not in every component you declare as router.